# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1523368447 14400
#      Tue Apr 10 09:54:07 2018 -0400
# Branch stable
# Node ID 6f1fde568e46360f76d99566856a97ff8326374b
# Parent  8cc3213f788f2a93e9689cddf4c00ba85649a95d
make .oct and .mex files in +package dirs work again (bug #51960)

* oct-parse.in.yy (load_fcn_from_file): Stash package name in oct and
mex function objects.

diff --git a/libinterp/parse-tree/oct-parse.in.yy b/libinterp/parse-tree/oct-parse.in.yy
--- a/libinterp/parse-tree/oct-parse.in.yy
+++ b/libinterp/parse-tree/oct-parse.in.yy
@@ -4748,16 +4748,18 @@ namespace octave
     if (len > 4 && file.substr (len-4, len-1) == ".oct")
       {
         if (autoload && ! fcn_name.empty ())
           nm = fcn_name;
 
         octave_function *tmpfcn
           = dyn_loader.load_oct (nm, file, relative_lookup);
 
+        tmpfcn->stash_package_name (package_name);
+
         retval = octave_value (tmpfcn);
       }
     else if (len > 4 && file.substr (len-4, len-1) == ".mex")
       {
         // Temporarily load m-file version of mex-file, if it exists,
         // to get the help-string to use.
 
         std::string doc_string;
@@ -4776,16 +4778,17 @@ namespace octave
           }
 
         octave_function *tmpfcn
           = dyn_loader.load_mex (nm, file, relative_lookup);
 
         if (tmpfcn)
           {
             tmpfcn->document (doc_string);
+            tmpfcn->stash_package_name (package_name);
 
             retval = octave_value (tmpfcn);
           }
       }
     else if (len > 2)
       {
         retval = parse_fcn_file (file, nm, dir_name, dispatch_type,
                                  package_name, true, autoload, autoload,
